src/index.ts

```diff
@@ -0,0 +1,154 @@
+import { ScraperEndpoint, ApolloEndpoint } from "./endpoints/index";
+// Contains the interface for the main template
+import { IMainTemplate } from "./templates/index";
+//#endregion
+
+//#region > Interfaces
+/**
+ * @interface Endpoint defines the structure of object a endpoint requirers to
+ *                     initialize.
+ */
+interface Endpoint {
+  /**
+   * Url: The URL of an endpoint. For performance reasons,
+   *      https should always be selected as protocol if possible.
+   */
+  url: string;
+  /**
+   * Type: It is possible to specify the type. This is currently only used to
+   *       differentiate multiple instances.
+   */
+  type: string;
+  headers: object;
+}
+
+/** @interface IClient will define the client structure */
+interface IClient {}
+//#endregion
+
+//#region > Classes
+/**
+ * @class The SNEK-client. Enjoy it. Will be implemented in the future
+ * @todo Rework the url checker and add documentation
+ */
+class Client implements IClient {
+  constructor(ep: Endpoint) {
+    /*
+     * When no protocol is defined, http will be appended. Therefore "https"
+     * should always be included for performance.
+     */
+    ep.url = ((url: string) => {
+      if (!/^(?:f|ht)tps?\:\/\//.test(url)) {
+        url = "http://" + url;
+      }
+
+      return url;
+    })(ep.url);
+  }
+}
+
+/** @class A client implementation for SNEK interaction */
+class SnekClient extends Client {
+  //#LEGACY
+  /** @deprecated Will be removed in the upcoming release */
+  endpoint: ApolloEndpoint;
+  gql: ApolloEndpoint;
+  template: IMainTemplate;
+  session: SnekSession;
+
+  /**
+   * @constructor
+   * @author Nico Schett <contact@schett.net>
+   * @param url The base URL the SnekClient should be working on.
+   *            Default: "https://engine.snek.at/api/graphiql".
+   * @param headers A object containing various request headers
+   * @param type A type description to differ between multiple instances
+   */
+  constructor(
+    url: string = "https://engine.snek.at/api/graphiql",
+    headers: object = {},
+    type: string = "graphql"
+  ) {
+    super({ type, url, headers });
+
+    this.template = new MainTemplate();
+    this.gql = new Apollo(url, { headers });
+    this.endpoint = this.gql;
+    this.session = new SnekSession("snek", this.gql, this.template.snek);
+  }
+}
+
+/** @class A client implementation for github interaction */
+class GithubClient extends Client {
+  //#LEGACY
+  /** @deprecated Will be removed in the upcoming release */
+  endpoint: ApolloEndpoint;
+  gql: ApolloEndpoint;
+  template: IMainTemplate;
+  session: GithubSession;
+
+  /**
+   * @constructor
+   * @author Nico Schett <contact@schett.net>
+   * @param url The base URL the GithubClient should be working on.
+   *            Default: "https://api.github.com/graphql".
+   * @param headers A object containing various request headers
+   * @param type A type description to differ between multiple instances
+   */
```

Comment:
Should have some sort of comment.

Initializes a Github client.

Reply:
Yeah, but not your suggestion cause this includes too less/useless information.
We have to discuss about a new TONIC rule in snek-at/tonic/issues/3